File: 2125-number-of-laser-beams-in-a-bank/2125-number-of-laser-beams-in-a-bank.java
class Solution {
    public int numberOfBeams(String[] bank) {
        int res = 0 , pre = 0;
        for(String s : bank)
        {
            int cnt = 0;
            for(int i = 0; i < s.length(); i++)
            {
                if(s.charAt(i) == '1')
                {
                    cnt++;
                }
            }
            if(cnt > 0)
            {
                res += cnt * pre;
                pre = cnt;
            }
        }
        return res;
    }
}

File: 2125-number-of-laser-beams-in-a-bank/2125-number-of-laser-beams-in-a-bank.cpp
class Solution {
public:
    int numberOfBeams(vector<string>& bank) {
        int sz  = bank.size(), pre(0), res(0);
        for(string &s : bank)
        {
            int cnt(0);
            for(char& ch : s)
            {
                if(ch == '1') cnt++;
            }
            if(cnt)
            {
                res += pre*cnt;
                pre = cnt;
            }
        }
        return res;
    }
};